support/vm/invocation_context.go

```diff
@@ -799,10 +853,7 @@ func (ic *invocationContext) replace(obj cbor.Marshaler) cid.Cid {
 	if !found {
 		ic.rt.Abortf(exitcode.ErrIllegalState, "failed to find actor %s for state", ic.msg.to)
 	}
-	c, err := ic.rt.store.Put(ic.rt.ctx, obj)
-	if err != nil {
-		ic.rt.Abortf(exitcode.ErrIllegalState, "could not save new state")
-	}
+	c := ic.StorePut(obj)
 	actr.Head = c
 	err = ic.rt.setActor(ic.rt.ctx, ic.msg.to, actr)
```

Comment:
We need to charge gas for the write of the actor struct into the state tree.

Reply:
This is intentionally not charging gas because as I understand the lotus vm code the filecoin protocol does not charge gas for reads and writes from the top level state tree.